phunware/maas-mapping-ios-sdk
# SWMAAS-1579 Update format of straight maneuvers with landmarks

IMPORTANT:
-
* Please read these instructions in their entirety before reviewing this PR.
* Please review [SWMAAS-1610](https://github.com/phunware/maas-mapping-ios-sdk/pull/132) before reviewing this PR, as this PR builds on top of it.
* You'll need the corresponding MapKit changes, which have not yet been merged into dev, so you'll need to use the `feature/SWMAAS-1579` branch of MapKit in your local working copy of MapKit, and modify the MapScenarios podfile to point at your working copy.

To test:
-
There are 2 patches we'll use, because we need to test 2 different scenarios for landmark-enabled instructions. Please apply each patch on top of this branch, and perform your testing.
* [This patch](https://github.com/phunware/maas-mapping-ios-sdk/files/4010492/swmaas_1579_long.patch.zip) can be used to test the "long" final maneuver.
* [This patch](https://github.com/phunware/maas-mapping-ios-sdk/files/4010487/swmaas_1579_short.patch.zip) can be used to test the "short" final manuever

Please test that:
* Landmark text/VO is used in the final step in the `Turn By Turn - Landmarks` example when the last manuever has a "long" distance
* Landmark text/VO is not used in the final step in the `Turn By Turn - Landmarks` example when the last manuever has a "short" distance
* The name of the POI should now be included in the final step for both the regular `Turn By Turn` example as well as `Turn By Turn - Landmarks` sample
* Everything is formatted according to the AC for this ticket

Testing Notes:
-
* The AC specifies 50 feet as the minimum distance for including landmark names in straight maneuvers (such as the final maneuver), however the current test data is much longer than this (since they actually entered it in meters in the portal). The patches included in this PR temporarily update the minimum value to 60 meters so that we can test that both cases (less than 50 feet and greater than 50 feet).
* The patches both have some temporary code to read instructions aloud using VO in `TurnByTurnViewController` and `TurnByTurnLandmarksViewController`. This code is not part of the PR, it is only included so that we can test that VO instructions are formatted correctly for both Basic and Landmark-enabled instructions.

Change Notes:
-
* If on a straight maneuver < 50 feet, do not display the landmark name in text/voiceover
* If on a straight manevuer > 50 feet, display the landmark name in text/voiceover
* For both basic and landmark-enabled instructions, the destination POI name should now be included in the last step
* Fixed several warning messages where the message displayed the incorrect file name
